drivers/SmartThings/sonos/src/init.lua

```diff
@@ -86,54 +88,66 @@ end
 -- a device is fully initialized whether we come from fresh join or restart.
 -- See: https://smartthings.atlassian.net/browse/CHAD-9683
 local function _initialize_device(driver, device)
-  if not device:get_field(PlayerFields._IS_INIT) then
-    log.trace(string.format("%s setting up device", device.label))
-    local is_already_found =
-    ((driver.found_ips and driver.found_ips[device.device_network_id])
-        or driver.sonos:is_player_joined(device.device_network_id)) and
-        driver._field_cache[device.device_network_id]
-
-    if not is_already_found then
-      log.debug("Rescanning for player with DNI " .. device.device_network_id)
-      local success = find_player_for_device(driver, device)
-      if not success then
-        device:offline()
-        log.error(string.format(
-          "Could not initialize Sonos Player [%s], it does not appear to be on the network",
-          device.label
-        ))
-        return
-      end
-    end
-
-    local fields = driver._field_cache[device.device_network_id]
-    driver._player_id_to_device[fields.player_id] = device -- quickly look up device from player id string
-    driver.sonos:mark_player_as_joined(fields.player_id)
-
-    log.trace("Setting persistent fields")
-    device:set_field(PlayerFields.WSS_URL, fields.wss_url, { persist = true })
-    device:set_field(PlayerFields.HOUSEHOULD_ID, fields.household_id, { persist = true })
-    device:set_field(PlayerFields.PLAYER_ID, fields.player_id, { persist = true })
-
-    device:set_field(PlayerFields._IS_INIT, true)
-  end
+  if not device:get_field(PlayerFields._IS_SCANNING) then
+    device:set_field(PlayerFields._IS_SCANNING, true)
+    cosock.spawn(
+      function()
+        if not device:get_field(PlayerFields._IS_INIT) then
+          log.trace(string.format("%s setting up device", device.label))
+          local is_already_found =
+          ((driver.found_ips and driver.found_ips[device.device_network_id])
+              or driver.sonos:is_player_joined(device.device_network_id)) and
+              driver._field_cache[device.device_network_id]
+
+          if not is_already_found then
+            log.debug("Rescanning for player with DNI " .. device.device_network_id)
+            device:offline()
+            local success = false
+
+            local backoff = utils.backoff_builder(360, 1, 0.5)
```

Comment:
Is there a reason to have randomness here? This is all hub local, right, so load shouldn't be a huge issue.

Reply:
We're introducing this change because we've seen cases where 3 minutes (which was the derived time limit based on other parameters) was insufficient to find a speaker, so we know that it could take a *very* long time.

While it's not gonna be the end of the world if we do this, say, every 30 seconds, we're starting to see cases where power users have a ton of load on their hubs via drivers so every little saving helps IMO.

Reply:
Sure, but that last parameter adjusts the backoff value randomly. I'm not understanding the advantage of randomly having a shorter or longer backoff has over a deterministic one.

Reply:
Copy/paste from another location, mostly, we can absolutely omit the randomization parameter here.

Reply:
@greens This has been adjusted. Thanks for the call-out!